src/pages/SearchPage.js

```diff
@@ -97,10 +107,16 @@ class SearchPage extends Component {
         if (option.reportID) {
             this.setState({
                 searchValue: '',
-            }, () => {
+            }, async () => {
                 if (this.props.isSmallScreenWidth) {
                     hideSidebar();
                 }
+                if (['web', 'desktop'].includes(getPlatform())) {
+                    modalHide();
+
+                    // Wait for modal animate out
+                    await new Promise(r => setTimeout(r, 330));
```

Comment:
There should not be any `setTimeout` in this solution, it was the main thing I was trying to avoid in the proposal.

Reply:
for this one,
If we want to see modal animation out then we need to wait for the animation to settle down.
because logically if we change the route before animate settles down then the modal will disappear from the DOM and we will unable to see the animation.

Reply:
I the master branch you can see animate out not work when we select another user rather than an already selected user.

Reply:
I thought that's what `animateOut` was for though?

Reply:
modals in HomePage handled through routes.
so with `animateOut`, this is the only solution that's I found
and without `animateOut` we can add a check on `RightDockedModal` like this way `route === currentURL`
Looking for your suggestions about this.